core/trino-main/src/main/java/io/trino/operator/scalar/DateTimeFunctions.java

```diff
@@ -308,34 +309,69 @@ public static long diffDate(@SqlType("varchar(x)") Slice unit, @SqlType(Standard
         return getDateField(UTC_CHRONOLOGY, unit).getDifferenceAsLong(DAYS.toMillis(date2), DAYS.toMillis(date1));
     }
 
+    /**
+     * Data tuple to store mapping between date time unit and date fraction extraction logic.
+     * This record is intended to be used for a specific use case only.
+     *
+     * @param unit date time unit represented as ASCII bytes
+     * @param dateTimeFieldProvider callback to extract fraction of date
+     */
+    private record DateTimeFieldProvider(byte[] unit, Function<ISOChronology, DateTimeField> dateTimeFieldProvider)
+    {
```

Comment:
You can create a secondary constructor that takes a String and does `.getBytes(US_ASCII)`

Reply:
Instead of keeping byte[] you can keep Slice instead

Reply:
Good idea! Added a secondary constructor.
As for the Slice usage, if has slightly worse performance, see the comment below.